src/bazel/bazel_query.ts

```diff
@@ -143,27 +149,83 @@ export class BazelQuery extends BazelCommand {
         `--output_base=${queryOutputBase}`,
       ]);
     }
-    const execOptions: child_process.ExecFileOptionsWithBufferEncoding = {
-      cwd: this.workingDirectory,
-      // A null encoding causes the callback below to receive binary data as a
-      // Buffer instead of text data as strings.
-      encoding: null,
-      maxBuffer: Number.MAX_SAFE_INTEGER,
-    };
-    try {
-      return (
-        await execFile(
-          this.bazelExecutable,
-          this.execArgs(options, additionalStartupOptions),
-          execOptions,
-        )
-      ).stdout;
-    } catch (e) {
-      if (ignoresErrors) {
-        return Buffer.alloc(0);
-      } else {
-        throw e;
+    return new Promise<Buffer>((resolve, reject) => {
+      // eslint-disable-next-line no-console
+      console.debug(
+        `Running Bazel query with command line: ${this.bazelExecutable} ${this.execArgs(
+          options,
+          additionalStartupOptions,
+        ).join(" ")}`,
+      );
+      const child = spawn(
+        this.bazelExecutable,
+        this.execArgs(options, additionalStartupOptions),
+        {
+          cwd: this.workingDirectory,
+          stdio: ["ignore", "pipe", "pipe"],
+        },
+      );
+
+      // Handle abort signal if provided
+      const onAbort = () => {
+        try {
+          child.kill();
+          reject(new Error("Query was aborted"));
+        } catch (error: unknown) {
+          reject(error);
+        }
+      };
+
+      const cleanup = () => {
+        if (abortSignal) {
+          // Type assertion to access the non-standard removeEventListener
+          const signal = abortSignal as unknown as {
+            removeEventListener: (type: string, handler: () => void) => void;
+          };
+          signal.removeEventListener("abort", onAbort);
+        }
+      };
+
+      if (abortSignal) {
+        if (abortSignal.aborted) {
+          onAbort();
+          return;
+        }
+        // Type assertion to access the non-standard addEventListener
+        const signal = abortSignal as unknown as {
```

Comment:
If you upgrade `@types/node` to latest you shouldn't have to do these type casts.